### PR TITLE
[WIP] Reimplement Facet to use the latest layout and headers

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: src/compile/facet.ts [hunk 1]
```diff
@@ -153,23 +154,70 @@ export class FacetModel extends ModelWithField {
     }];
   }
 
-  public parseAxis() {
-    this.child.parseAxis();
-    // FIXME: Correct write parseAxis logic
-    this.component.axes = {};
+  // TODO: rename to parseAxisAndHeaders
```
Comment (domoritz) [inline src/compile/facet.ts]: done

File: src/compile/layout/header.ts [hunk 2]
```diff
@@ -1,65 +1,138 @@
...
+  if (header) {
+    let title = null;
+    if (layoutHeader.field && header.labels) {
+      title = {
+        text: {signal: `parent['${layoutHeader.field}']`},
+        offset: 10,
+        orient: channel === 'row' ? 'left' : 'top',
+        encode: {
+          update: {
+            fontWeight: {value: 'normal'},
```
Comment (domoritz) [inline src/compile/layout/header.ts]: so this is the hack we talked about today with jeff?

**Comment by domoritz** [comment]
LGTM

File: src/compile/layout/header.ts (hunk 2 above)
Comment (kanitw) [inline src/compile/layout/header.ts]: Yes.

**Pull request closed without merging**